### PR TITLE
Added chinese remainder theorem in C.

### LINKED ISSUE #2481: Chinese Remainder Theorem
- [x] C - @Anushka-Rajvanshi
- [x] C++
- [x] C#
- [x] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [x] Go - @jedicakelord
- [x] Java
- [x] JS
- [ ] Kotlin
- [x] Php - @sukritishah15
- [x] Python
- [x] Readme - @chandrikadeb7
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Anushka-Rajvanshi**
Fixes #2481 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added implementation of chinese remainder theorem in C.
- 

#### Languages Used:
- C
- 

#### Files Added:
- 
[chinese_remainder_theorem.docx](https://github.com/jainaman224/Algo_Ds_Notes/files/4431132/chinese_remainder_theorem.docx)

- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Anushka-Rajvanshi** [comment]
@jainaman224  @raghav-dalmia @somya-kapoor  please review my PR.

File: Chinese_Remainder_Theorem/chinese_remainder_theorem.c [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    do
+    {  
+        if((res%divisor[i]) == rem[i])
+        {
+ // When value of i is equal to value of size i.e. all remainder matches with given rem[] array, then res is the minimum value
+            i++ ;
+        }
+        else
+        {    //if above condition gives false for any value of i, check for next number.
+            res++ ;
```
Comment (raghav-dalmia) [inline Chinese_Remainder_Theorem/chinese_remainder_theorem.c]: Your code gives a brute force solution. It's not CRT.

Comment (Anushka-Rajvanshi) [reply Chinese_Remainder_Theorem/chinese_remainder_theorem.c]: Sir this method is also used in code implemented using cpp so I thought this method is acceptable.

Comment (Anushka-Rajvanshi) [reply Chinese_Remainder_Theorem/chinese_remainder_theorem.c]: @raghav-dalmia

Comment (raghav-dalmia) [reply Chinese_Remainder_Theorem/chinese_remainder_theorem.c]: It will give the same output, but that's not CRT.

Comment (Anushka-Rajvanshi) [reply Chinese_Remainder_Theorem/chinese_remainder_theorem.c]: Ok sir. Will try to implement CRT.

Comment (raghav-dalmia) [reply Chinese_Remainder_Theorem/chinese_remainder_theorem.c]: Change your code. It's not acceptable for CRT. You can create a different issue for this solution(if it's not present).

**Comment by raghav-dalmia** [comment]
are you working on it?

**Comment by Anushka-Rajvanshi** [comment]
Yes sir. Give me just one more day. @raghav-dalmia

**Comment by Anushka-Rajvanshi** [comment]
@raghav-dalmia Please review this PR .

**Comment by raghav-dalmia** [comment]
Squash your commits, rest looks good to me.

File: Chinese_Remainder_Theorem/chinese_remainder_theorem.c [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        {
+            if(((M[i] % divisor[i]) *  k) % divisor[i] == 1)
+            {
+                inv[i] = k;
+                count++;
+            } 
+            k++; 
+        }while(count == 0);  
+    }
+    for(i = 0; i < size; i++)
```
Comment (MastersAbh) [inline Chinese_Remainder_Theorem/chinese_remainder_theorem.c:29]: please add space after for across the file

Comment (Anushka-Rajvanshi) [reply Chinese_Remainder_Theorem/chinese_remainder_theorem.c:29]: Sorry. I have made the requested changes but I am unable to squash the commits. So I am making a new PR.

Comment (Anushka-Rajvanshi) [reply Chinese_Remainder_Theorem/chinese_remainder_theorem.c:29]: Please review PR #2832

**Pull request closed without merging**